### DISCUSSION TITLE
TabControl

### DISCUSSION THREAD

**Discussion by gHensges** [AvaloniaUI/Avalonia#8770] [Q&A] [answered]
Hello,

I am new to Avalonia, so I do not have much experiences.

At the moment I am trying to create a custom header to be shown inside the tabitem control, because I believe that the Standard is not looking as I am expecting. But I do not know what to override. I was googling around and found nice examples of what to do using WPF but not a single example for Avalonia. The WPF is definitely not working.
Does anybody has an example of a TabControl with a customized TabItem header (I want to show an Icon together with some Text).

**Comment by BAndysc** [comment, marked as answer]
Contrary to WPF, with Avalonia it is very easy to read the source code of original styles, so let's take a look at the TabControl style:
https://github.com/AvaloniaUI/Avalonia/blob/release/0.10.18/src/Avalonia.Themes.Fluent/Controls/TabControl.xaml#L42

TabControl consists of an ItemPresenter - which displays the tabs and a ContentPresenter which displays the current tab. ItemPresenter's ItemTemplate is set to `{TemplateBinding ItemTemplate}`. So you want to modify `TabControl.ItemTemplate`. That's all

```
<TabControl>
    <TabControl.ItemTemplate>
        <DataTemplate> <!-- any data template you want, this is just an example -->
            <StackPanel>
                <Image Source="{Binding Icon}" Width="16" Height="16" />
                <TextBlock Text="{Binding Title}" />
            </StackPanel>
        </DataTemplate>
    </TabControl.ItemTemplate>
</TabControl>
```

**Reply by gHensges** [reply]
Thanks for your quick response. Works well!
